hack/dockerfiles/freebsd-smoke.Dockerfile

```diff
@@ -0,0 +1,3 @@
+FROM dougrabson/freebsd-minimal:13.1
```

Comment:
I guess this is only used in CI, but _slightly_ uncomfortable with using a (from the looks of it) random freebsd image. I wonder if there's ever been any attempts on getting "official" freebsd image on Docker Hub; do you know, @tianon

Reply:
cc @dfr

Reply:
I am responsible for this image and I am a member of the FreeBSD project (for a really long time). I would really like there to be a project-owned image but last time I checked, both docker.io/freebsd and quay.io/freebsd seem to be taken. I will try to start a conversation in the project on the best way forward here.

Reply:
> I am responsible for this image and I am a member of the FreeBSD project (for a really long time)

Oh, thanks for the extra info!

>  both docker.io/freebsd and quay.io/freebsd seem to be taken

Hm, right. So (slightly confusing) if it would be an official image, the reference would be `docker.io/library/freebsd` (the `library/` namespace is what's used for the official images, so if you would do `docker run freebsd`, the canonical name is `docker.io/library/freebsd`).

I see the https://hub.docker.com/u/freebsd looks empty (but of course there may be activity on the account). I can reach out to the team on Docker Hub to inform if that account was registered by a user, or if it was pre-emptively registered by the Docker Hub team to prevent it from being used.

Let me see if I can find the right people to see

a) if there's an option for the FreeBSD maintainers  to  get ownership of that account, and
b) what it would take to potentially have freebsd included in `library/` (to become an official image) - this may have further implications (should other official images now also have a freebsd variant etc?), checking never hurts though, so let me try.

Reply:
> > I am responsible for this image and I am a member of the FreeBSD project (for a really long time)
> 
> Oh, thanks for the extra info!
> 
> > both docker.io/freebsd and quay.io/freebsd seem to be taken
> 
> Hm, right. So (slightly confusing) if it would be an official image, the reference would be `docker.io/library/freebsd` (the `library/` namespace is what's used for the official images, so if you would do `docker run freebsd`, the canonical name is `docker.io/library/freebsd`).
> 
> I see the https://hub.docker.com/u/freebsd looks empty (but of course there may be activity on the account). I can reach out to the team on Docker Hub to inform if that account was registered by a user, or if it was pre-emptively registered by the Docker Hub team to prevent it from being used.
> 
> Let me see if I can find the right people to see
> 
> a) if there's an option for the FreeBSD maintainers to get ownership of that account, and b) what it would take to potentially have freebsd included in `library/` (to become an official image) - this may have further implications (should other official images now also have a freebsd variant etc?), checking never hurts though, so let me try.

I would be very interested in the answers to both (a) and (b) - thanks for the help!

Reply:
I put out an internal question;

- Didn't get an answer (yet) on A) - but will let you know once I do (person I asked is in the USA, so may not be around yet, and it's nearing the Holidays, so some people may be on PTO
- For B), this would indeed have a bit more impact, and it looks like the preference is to first get some stability on an FreeBSD image ("A" could be a stepping-stone towards that) before the official images can commit to including FreeBSD as options for the official images.

Let's see if we can get "A" in motion though; I'll try to keep you posted. In the meantime, perhaps on your side it would be possible to discuss with the FreeBSD community to see who would be the canonical contact(s) for this.

Reply:
I am happy to be a contact for this - either via dfr@rabson.org or dfr@freebsd.org. It would probably be best to also copy our release engineering group via re@freebsd.org

Reply:
Thanks! I'll forward that information to the internal thread I started, in case they want to reach out 👍

Reply:
There's some additional discussion about this topic here: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=254645

It'd be good to have an image maintained by the FreeBSD project.  The immediate main barrier for this is: the majority of the `library/` images are built with [bashbrew](https://github.com/docker-library/bashbrew#readme), which depends on a running copy of Docker on the target platform.  @tianon might have some thoughts on whether bashbrew can be modified to support FreeBSD's needs or if a different tool could be used.

In the meantime, [runj](https://github.com/samuelkarp/runj) can generate compatible images based on the standard FreeBSD `base.txz` archives.  If the FreeBSD project can gain control over https://hub.docker.com/u/freebsd, https://quay.io/organization/freebsd, or https://gallery.ecr.aws/freebsd (appears to be unclaimed) that can provide another location for images to be published until the `library/`/bashbrew issue is resolved.

Reply:
> It'd be good to have an image maintained by the FreeBSD project. The immediate main barrier for this is: the majority of the library/ images are built with bashbrew, which depends on a running copy of Docker on the target platform. @tianon might have some thoughts on whether bashbrew can be modified to support FreeBSD's needs or if a different tool could be used.
 
Ah, yes, didn't even consider the bashbrew side of things. So yes, having images FreeBSD images inside a `freebsd` namespace (`docker pull freebsd/freebsd` (e.g.)) could be a good start (and I _expect_ the org would qualify for the "Sponsored OSS" programme). I'll keep an eye; as mentioned, it's the Holiday season, so perhaps replies will arrive sometime after the Holidays.

Reply:
Very short update on the above; I got some traction on this internally last Friday, and a colleague will set up an e-mail thread to help coordinate things.

Reply:
We had an email thread with re@freebsd.org, and they weren't comfortable having an image published in such an official place that they weren't creating/maintaining directly, so while I'm not a BuildKit maintainer, I think this image maintained by Doug is probably the most official we're going to get for now and IMO this concern doesn't make sense to block this PR on. :bow: